devices/debian_wifi.py

```diff
@@ -17,3 +18,54 @@ def scan(self):
         self.expect(self.prompt)
 
         return self.before
+
+    def wpa_connect(self,ssid_name,pass_word,wlan_intf):
+
+         '''Generate WPA supplicant file and execute it'''
+
+         self.sendline("rm /etc/"+ssid_name+".conf")
+         self.expect(self.prompt)
+         self.sendline("wpa_passphrase "+ssid_name+ " >> /etc/"+ssid_name+".conf")
+         self.expect("")
+         self.sendline(pass_word)
+         self.expect(self.prompt)
```

Comment:
Just a note, i don't really know what the wpa_passphrase does, and it is possible that this  command, NEVER fails. But let's say something has gone wrong. at the moment you are only expecting the prompt, i.e. assuming everything is fine. In this case you could check the value of "$?", if == 0 the command succeeded, !=0 something has failed. 
Would this be applicable in this situation?

Reply:
In this case we are just passing the password and creating a wpa_passphrase config file. Once the password is provided, it will end up in the prompt. Probably I will add a timeout of 20 sec for expecting the prompt.